src/lib.rs

```diff
@@ -226,7 +226,7 @@ pub enum SchemeData {
 }
 
 /// Components for URLs in a *relative* scheme such as HTTP.
-#[derive(PartialEq, Eq, Clone, Debug, Hash, PartialOrd, Ord)]
+#[derive(Clone, Debug, Hash, PartialOrd, Ord)]
```

Comment:
Why not? Return `hash(self) == hash(other)` from `eq` method.

Anyway, thanks for the quick review, I'll get to work.

On Mon, Nov 02, 2015 at 01:15:37PM -0800, Manish Goregaokar wrote:
> > @@ -226,7 +226,7 @@ pub enum SchemeData {
> >  }
> >  
> >  /// Components for URLs in a *relative* scheme such as HTTP.
> > -#[derive(PartialEq, Eq, Clone, Debug, Hash, PartialOrd, Ord)]
> > +#[derive(Clone, Debug, Hash, PartialOrd, Ord)]
> 
> Nope. It doesn't, and it's not possible to either, since there isn't enough information on what choices to make. In general an unsolvable problem.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/servo/rust-url/pull/133/files#r43683364